classes/i18n/Locale.php

```diff
@@ -365,54 +365,38 @@ public function getScripts(?string $locale = null): Scripts
     /**
      * Get the formatted locale display names with country if same language code present multiple times
      * 
-     * @param   array $filterByLocales  Optional list of locales code to filter by the returned formatted names list
-     * @param   array $locales          Optional list of availabel all locales
-     * @param   int   $langLocaleStatus The const value of one of LocaleMetadata:LANGUAGE_LOCALE_*
+     * @param   array $filterByLocales          Optional list of locales code to filter by the returned formatted names list
+     * @param   array $locales                  Optional list of availabel all locales
+     * @param   int   $langLocaleStatus         The const value of one of LocaleMetadata:LANGUAGE_LOCALE_*
+     * @param   bool  $omitLocaleCodeInDisplay  Should leave out the locale code from display. By deault leave out.
      * 
-     * @return  array                   The list of locales with formatted display name
+     * @return  array                           The list of locales with formatted display name
      */
-    public function getFormattedDisplayNames(array $filterByLocales = null, array $locales = null, int $langLocaleStatus = LocaleMetadata::LANGUAGE_LOCALE_WITH): array
+    public function getFormattedDisplayNames(array $filterByLocales = null, array $locales = null, int $langLocaleStatus = LocaleMetadata::LANGUAGE_LOCALE_WITH, bool $omitLocaleCodeInDisplay = true): array
     {
         $locales ??= $this->getLocales();
 
+        if ($filterByLocales) {
```

Comment:
Since we're using `null` as a special case value signifying "don't filter anything", I'd suggest making this an exact `=== null` check. This will ensure that the behaviour doesn't get accidentally applied to `[]` as well.